### PR TITLE
Improve Streamlink documentation

### PR DESCRIPTION
**Pull request by fraxxio**
Solving #5073
Added new QLabel in ExternalToolsPage.cpp.

Note

### CODE REVIEW

File: src/widgets/settingspages/ExternalToolsPage.cpp [hunk 1]
```diff
@@ -60,6 +60,12 @@ ExternalToolsPage::ExternalToolsPage()
                                  getSettings()->streamlinkUseCustomPath);
         groupLayout->setWidget(2, QFormLayout::SpanningRole, customPathCb);
 
+        auto note =
+            new QLabel("The custom path may not need the file extension.");
+        note->setWordWrap(true);
+        note->setStyleSheet("color: #bbb");
+        groupLayout->setWidget(3, QFormLayout::SpanningRole, note);
```
Comment (Mm2PL) [inline src/widgets/settingspages/ExternalToolsPage.cpp:67]: This is not accurate on Linux. To create a new process on Linux you *have* to have the full file name, extensions are not special.

**Review by Mm2PL — CHANGES_REQUESTED** [review]
Doesn't apply to Linux and I don't think it applies to Mac either. Should probably be if-defed to Windows only.

File: src/widgets/settingspages/ExternalToolsPage.cpp (hunk 1 above)
Comment (fraxxio) [reply src/widgets/settingspages/ExternalToolsPage.cpp:67]: I misunderstood the issue, the problem is not with the custom path but with the name of the executable. I was able to recreate the issue #5073 on linux, where renaming the executable name from `streamlink.AppImage` to `streamlink` fixed the problem. 
So the note message should probably be `If Chatterino is not able to find Streamlink executable, renaming the executable to not have a file extension might help.`, or something like that. Also this message probably should only appear on Linux.

Comment (pajlada) [reply src/widgets/settingspages/ExternalToolsPage.cpp:67]: It's probably best to say what Chatterino looks for instead, something like "Chatterino expects the binary to be called X" where X should be the name of the streamlink binary depending on the platform

File: src/widgets/settingspages/ExternalToolsPage.cpp [hunk 2]
```diff
@@ -60,6 +69,12 @@ ExternalToolsPage::ExternalToolsPage()
                                  getSettings()->streamlinkUseCustomPath);
         groupLayout->setWidget(2, QFormLayout::SpanningRole, customPathCb);
 
+        auto note = new QLabel("Chatterino expects the binary to be called ");
+        note->setText(note->text().append(getBinaryName()));
```
Comment (Nerixyz) [inline src/widgets/settingspages/ExternalToolsPage.cpp:73]:
```suggestion
        auto note = new QLabel(QStringLiteral("Chatterino expects the binary to be called \"%1\".").arg(STREAMLINK_BINARY_NAME);
```

We can set the text correctly from the start.

File: src/widgets/settingspages/ExternalToolsPage.cpp [hunk 3]
```diff
@@ -14,6 +14,15 @@
...
 namespace chatterino {
 
+const char *getBinaryName()
+{
+#ifdef _WIN32
+    return "streamlink.exe";
+#else
+    return "streamlink";
+#endif
+}
```
Comment (Nerixyz) [inline src/widgets/settingspages/ExternalToolsPage.cpp:24]: This should be in an anonymous namespace (`namespace { }`) above. Or, even better imo: create a `constexpr QStringView STREAMLINK_BINARY_NAME = u"abc";`

Comment (pajlada) [reply src/widgets/settingspages/ExternalToolsPage.cpp]: @Nerixyz Could you give your opinion on how it's done here [`4e19758` (#5076)](https://github.com/Chatterino/chatterino2/pull/5076/commits/4e19758c047e368690a1d17ce05c2b895427da3b) ?

**Review by pajlada — APPROVED** [review]
Looks good to me, would like approval from the previous reviewers before merging this in

I took the opportunity to do some minor refactors at the same time, thank you for your contribution!

**Review by Mm2PL — APPROVED** [review]

**Review by Nerixyz — APPROVED** [review]

**Pull request merged**